src/pybind/mgr/cephadm/services/cephadmservice.py

```diff
@@ -73,6 +75,61 @@ def simplified_keyring(entity: str, contents: str) -> str:
     return keyring
 
 
+def get_dashboard_eps(svc: 'CephadmService') -> Tuple[List[str], Optional[str]]:
+    dashboard_eps: List[str] = []
+    port = None
+    protocol = None
+    mgr_map = svc.mgr.get('mgr_map')
+    url = mgr_map.get('services', {}).get('dashboard', None)
+    if url:
```

Comment:
I thought that the mgr API had some direct method to fetch the service addresses 🤔

Reply:
I've just moved this code from other place. I'm not aware of a better API if there's some I can change the code to make use of it :+1: